src/pod-security-policy/seccomp/src.cel

```diff
@@ -0,0 +1,186 @@
+    })
+- name: containerAnnotationsProfiles
+  expression: |
+    variables.badContainers.filter(container, 
+      !(has(container.securityContext) && has(container.securityContext.seccompProfile)) && 
+      !variables.hasPodSecurityContext && 
+      has(variables.anyObject.metadata.annotations) && (("container.seccomp.security.alpha.kubernetes.io/" + container.name) in variables.anyObject.metadata.annotations)
+    ).map(container, {
+      "container" : container.name,
+      "profile" : variables.anyObject.metadata.annotations["container.seccomp.security.alpha.kubernetes.io/" + container.name],
+      "file" : dyn(""),
+      "location" : dyn("annotation container.seccomp.security.alpha.kubernetes.io/" + container.name),
+    })
+- name: podLocalHostProfile
+  expression: |
+    has(variables.anyObject.spec.securityContext) && has(variables.anyObject.spec.securityContext.seccompProfile) && has(variables.anyObject.spec.securityContext.seccompProfile.localhostProfile) ? variables.anyObject.spec.securityContext.seccompProfile.localhostProfile : ""
+- name: podSecurityContextProfiles
+  expression: |
+    variables.badContainers.filter(container, 
+      !(has(container.securityContext) && has(container.securityContext.seccompProfile)) && 
+      variables.hasPodSecurityContext
+    ).map(container, {
+      "container" : container.name,
+      "profile" : variables.anyObject.spec.securityContext.seccompProfile.type,
+      "file" : variables.podLocalHostProfile,
+      "location" : dyn("pod securityContext"),
+    })
+- name: containerSecurityContextProfiles
+  expression: |
+    variables.badContainers.filter(container, 
+      has(container.securityContext) && has(container.securityContext.seccompProfile)
+    ).map(container, {
+      "container" : container.name,
+      "profile" : container.securityContext.seccompProfile.type,
+      "file" : has(container.securityContext.seccompProfile.localhostProfile) ? container.securityContext.seccompProfile.localhostProfile : dyn(""),
+      "location" : dyn("container securityContext"),
+    })
+- name: containerProfilesMissing
+  expression: |
+    variables.badContainers.filter(container, 
+      !(has(container.securityContext) && has(container.securityContext.seccompProfile)) && 
+      !(has(variables.anyObject.metadata.annotations) && (("container.seccomp.security.alpha.kubernetes.io/" + container.name) in variables.anyObject.metadata.annotations)) && 
+      !variables.hasPodSecurityContext && 
+      !variables.hasPodAnnotations 
+    ).map(container, {
+      "container" : container.name,
+      "profile" : dyn("not configured"),
+      "file" : dyn(""),
+      "location" : dyn("no explicit profile found"),
+    })
+- name: allBadContainerProfiles
+  expression: |
+    variables.podAnnotationsProfiles + variables.containerAnnotationsProfiles + variables.podSecurityContextProfiles + variables.containerSecurityContextProfiles + variables.containerProfilesMissing
```

Comment:
Kubernetes no longer reads security contexts from the annotation, we should give constraint authors the ability to turn off relying on the annotation in order to avoid a security hole where pod runners "pretend" to have the correct security context.

I.e. we should add a parameter to turn off reading from the annotation.

Here is the k8s code for retrieving the seccomp profile:

https://github.com/kubernetes/kubernetes/blob/c6b5191c37f939d2d61e76de222a96ae5f5d9558/pkg/kubelet/kuberuntime/helpers.go#L270-L291

Reply:
Added v2 for seccomp with a variable to enable/disable reading from annotations.

Reply:
If we are creating "v2", we should just remove annotations from v2 altogether, since K8s no longer supports seccomp annotations